### PR TITLE
Text::preg_replace_array()

### PR DESCRIPTION
**Pull request by robbytaylor**
Following on from https://github.com/kohana/core/pull/277#issuecomment-7102571

A Text helper to perform an array of replacements on a string.

### CODE REVIEW

**Comment by shadowhand** [comment]
Not sure I like the name of the method, but the functionality is solid.

**Comment by robbytaylor** [comment]
I don't much like the name either to be honest but I figured it describes what the method does, and I couldn't think of anything which made it clearer.

File: classes/kohana/text.php [hunk 1]
```diff
@@ -603,4 +603,22 @@ public static function widont($str)
 		return $str;
 	}
 
+	/**
+	 * Performs an array of regular expression replacements on a string.
+	 *
+	 * @param	string	$text	The text for perform replacements on.
+	 * @param	array	$regex	Array of regular expression replacements in the form $pattern => $replacement
+	 * @param	int		$limit 	The maximum number of replacements for each pattern/replacement in $replacements
```
Comment (zeebinz) [inline classes/kohana/text.php]: Should be: $replacements, integer

File: classes/kohana/text.php [hunk 2]
```diff
@@ -603,4 +603,22 @@ public static function widont($str)
 		return $str;
 	}
 
+	/**
+	 * Performs an array of regular expression replacements on a string.
+	 *
+	 * @param	string	$text	The text for perform replacements on.
+	 * @param	array	$replacements	Array of regular expression replacements in the form $pattern => $replacement
+	 * @param	integer		$limit 	The maximum number of replacements for each pattern/replacement in $replacements
```
Comment (evanpurkhiser) [inline classes/kohana/text.php]: The spacing is all wonky here. They should probably be aligned.

File: classes/kohana/text.php [hunk 3]
```diff
@@ -603,4 +603,22 @@ public static function widont($str)
...
+	 * Performs an array of regular expression replacements on a string.
+	 *
+	 * @param	string	$text			The text for perform replacements on.
+	 * @param	array	$replacements	Array of regular expression replacements in the form $pattern => $replacement
+	 * @param	integer	$limit 			The maximum number of replacements for each pattern/replacement in $replacements
+	 * @return	string
+	 */
+	public static function preg_replace_array($text, array $replacements, $limit = -1)
+	{
+		foreach ($replacements as $pattern => $replacement)
```
Comment (cs278) [inline classes/kohana/text.php:616]: `return preg_replace(array_keys($replacements), $replacements, $text, $limit);`

**Comment by zeelot** [comment]
Could you make a ticket for this please?

**Comment by zeelot** [comment]
What @cs278 said seems to indicate that we don't need a method like this. Let me know if I missed something.

**Pull request closed without merging**